packages/flutter_tools/test/integration.shard/hot_reload_test.dart

```diff
@@ -16,6 +16,10 @@ import 'test_driver.dart';
 import 'test_utils.dart';
 
 void main() {
+  testAll();
```

Comment:
I think it'd be better if we kept the test logic in a separate file (e.g., `hot_reload_test_common.dart`) and have the runtime specific test files both import the file.

Also, I'd wrap each invocation of `testAll()` with a `group` that has a name that includes the runtime target so that it'll be more clear which version of the test is running.

Reply:
> have the runtime specific test files both import the file

The Flutter analyze bots gave me a friendly reminder to do so as well. :D I'll have to move this to `test_data` to avoid a separate analysis issue: https://github.com/flutter/flutter/blob/385878e7c1448df4762a7a7f99a662bb3c5d2ed4/dev/bots/analyze.dart#L561.

Done, and modified the restart tests to be consistent here as well.